WIP Add comment field in File web page

Oyabi opened the pull request:
### Changes description

Add a comment field in File webpage and File list view.
Useful in the case where several files have the same name in order to differentiate them.

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### References

<!-- issues related (for reference or to be closed) and/or links of discuss -->

Closes #N/A

btry commented on install/mysql/plugin_flyvemdm_empty.sql:
```diff
@@ -259,3 +260,8 @@ CREATE TABLE `glpi_plugin_flyvemdm_taskstatuses` (
+-- glpi_displaypreferences
+INSERT INTO `glpi_displaypreferences` (`id`, `itemtype`, `num`, `rank`, `users_id`)
+   VALUES (NULL,'PluginFlyvemdmFile', '1', '1', '0'),
+          (NULL,'PluginFlyvemdmFile', '4', '2', '0');
```
Hi

This query will fail if for some reason the preference already exists, because of the unicity of (users_id, itemtype, num).

I think this is better to create  the row from PHP in the installation code.

btry commented on install/upgrade/update_to_dev.php:
```diff
@@ -205,6 +205,19 @@ function plugin_flyvemdm_update_to_dev(Migration $migration) {
+                VALUES (NULL, 'PluginFlyvemdmFile', '1', '1', '0'),
+                (NULL, 'PluginFlyvemdmFile', '4', '2', '0');";
+      $DB->query($query);
+   }
```
Related to my previous comment, let's move the code to create the display preference in a method called for both install and upgrade process.

btry commented on inc/file.class.php:
```diff
@@ -144,6 +144,10 @@ public function prepareInputForAdd($input) {
+      if (!isset($input['comment']) || empty($input['comment'])) {
+         $input['comment'] = '';
+      }
+
```
I think this block is not necessary. If comment is not populated, then GLPI will not try to update it.

DIOHz0r commented on composer.lock:
```diff
@@ -0,0 +1,3989 @@
+{
```
This file is not necessary on git, please remove it

Oyabi commented:
Hi @ajsb85,

I don't know why but it seems I can't change the commit message.
I try that command : `git rebase -i HEAD~5` (and other number) but my commit message do not appear in the list.
Have you a solution for me ?

Regards.

btry commented:
@Oyabi Do you see at least one commit in the list ?

Also, I usually use a caret instead a tilde when choosing a commit backwards from HEAD. Give it a try.

Oyabi commented:
Yes i see many commit:
```
pick 2fc6c4d feat(file): add comment field
pick c682e80 style(agent): indentation fix
pick 2e7362d test: use version 1.0 of test package
pick 1e7e553 fix(package): improve UX by allowing to send package name
pick dd908fd fix(agent): reset device enroll status on delete (issue #32)
pick cd01ad2 fix(agent): removed action of delete computer on deletion
pick 93c2bf9 test(agent): added check for no deletion of computer
pick 098b992 feat(agent): updated code to import computer inventory with F.I.
pick 3fafa99 test(enrollment): after login, GLPI needs to be reloaded again
pick bb480c7 test(install): update configuration settings for plugin
pick 5b761ea test(invitation): update code for assert session message
pick 2b26df2 test(agent): update code for test agent with FI
pick 4357f80 test(inventory): refactor code for multiple uses of XML inventory
pick dd1ee9d test: fixed tests that are using the TestingCommonTools
pick a7e5ebb test: fixed php error exit 255 on tests.
pick 8702f1c test(invitation): fixed code unit test for invitations in any machine
pick 3b87d7e test(agent): fixed invalid inventory string for login agent
pick 6b40cc0 test(agent): fix XML import notice
pick c155017 test(agent): fixed test for agent integration
pick 386ceb9 style: updated code for coding standards
pick 81207ae test(package): fixed test integration for package
pick 6554cbe chore(travis): normalize travis
pick 7786567 test: fix missing incldue
pick da06c7d test: normalize namespace specification of classes
pick 4e3e245 test: add a patch for GLPI to workaround a bug with current tests.
pick 3b657c0 chore(travis): speed up tests with a ramdisk
pick d5068f3 test: try to rely on psr autoload
pick 6fafbea docs: change name of author
pick 7c8e9df chore: improve build tool
pick 9096104 fix(agent): update search options
pick ce5947f fix(agent): non conform definition of booleans
pick 837642a test: remove quite useless unit  test
pick 92ac441 docs: update headers
pick 4f71481 feat(install): template for installation wizard
pick 1b93672 docs(config): updated year on header
pick a73a1ed fix(agent): xml of inventory is now decoded
pick 67e5aa4 fix(agent): redirect output from FI messages on enrollment
pick 89bc793 fix(agent): removed base64 encoding for XML inventory
pick 5399fde fix(config): fix errors in wizard (#263)
pick 0ff79c1 fix(agent): updated code for not register invalid user agents
pick 8f6b5d9 fix(agent): updated code for reload FI plugin
pick 31018f9 fix(agent): fixed XML inventory request, it must be on base64
pick 7f52837 docs(wizard): updated install wizard text on mosquitto set-up
pick 93e94c7 fix(search): update search options fields for Flyvemdm ItemTypes
pick e0314b9 docs(plugin): updated plugin.xml info
pick b030801 docs(plugin): fixed plugin.xml info
pick 1f78037 fix(wizard): improve text of the wizard
pick d966a17 fix(wizard): typo
pick a167169 feat(message): remove queing of mqtt messages and grouping
.
.
.
pick 5a989aa fix(task): non unique tasks overwrite each other
pick 73dbc94 fix(agent): some mqtt topics not cleaned up
pick 670bb6f feat(mqtt): debug mode for daemon
pick 22e21ff ci: introduce script
pick 2a92e14 fix: typo
pick 6318278 test: fix topics to cleanup
pick cc09c81 fix(file): accessibility issue
pick a818a7f fix(comment): Remove useless code
pick e95c0bf fix: Remove useless file
```
As you can see, my commit message is absent from that list.
Same result with `^`.
Regards.

btry commented:
That's strange, I use this command quite often, without such problem. 

I'll checkout your branch, and try myself this afternoon. I'll let you know the result.

btry commented:
Hi again @Oyabi 

I think your code is based on a quite old revision of the develop branch. I think there are some other details I don't know about the status of your clone when you began to code.

FYI, here is what I did to solve your issue:
- git checkout develop
- git pull origin develop
- git checkout -b Oyabi-develop_2 develop
- git cherry-pick 2fc6c4df7c2da77b5a8a5c2890831870e4f25b47 010046d40593eafa41dc0313374deb63d862a551 cc09c81bc1f2f08dc903fc6de9e9de990adfa683 a818a7f8b00099bd7936fa82a88f1c103b1a9bfe e95c0bf5dfffd64b844759f894ce636331c98a4e
This command picks all your commits you did, and applies them on the current branch Oyabi-develop_2
I had to resolve some conflicts, showing our code is based on a old version of the project.
- git diff develop..
this command shows a diff from the branch develop to the curent branch (containing your contributions). I usually this way the cherry-pick is as expected.

Once done, a git rebase -i should work as expected.

You should always create a new branch when trying to write some code:
- git checkout develop
- git pull origin develop
- git checkout -b feature/my_super_feature develop
- start to code, commit...
- git push -u Oyabi

In all the commands I assume the remote **origin** is the official repository and **Oyabi** is your fork. 

I recommend you use git flow to better manage your branches. This workflow is rather simple and helps you with mangement of your copy of a project. It is implemetned with some CLI tools and many IDE support it also.

I pushed here the result of what I described above:
https://github.com/btry/glpi-plugin/tree/Oyabi-develop_2

btry commented:
Hi @Oyabi 

Today I testes your contribution in a functional way. I found an issue (not due to your code), I changed a bit the template for the UI, and I fixed for you the changes I requested.

I merged all commits into a single one, for a cleaner history. Look again the branch on my repo. 

I'll merge it soon, after checking you're still the author of the code.

btry commented:
Hi @Oyabi 

I created a new PR #328 based on your contribution. I close this PR, and suggest you follow the new one. I'll merge it as soon unit test pass. 

Thank you for your contribution.

btry commented:
@Oyabi : hi, your contribution is merged. Enjoy !

Oyabi commented:
I'm sorry for my late response but thank you very much @btry for your time and work on my push.
I will keep your comment on git command on my bookmark.

Regards.
